src/plot_api/plot_api.js

```diff
@@ -2092,9 +2092,6 @@ function _relayout(gd, aobj) {
     var oldWidth = gd._fullLayout.width,
         oldHeight = gd._fullLayout.height;
 
-    // coerce the updated layout
-    Plots.supplyDefaults(gd);
```

Comment:
Yeah, this is ridiculously complicated... actually tracing through it I found a very specific new bug, though not sure yet if it's exactly related. But I think we can sort it out without reinstating that redundant `supplyDefaults`:

All of the pathways after `_relayout` in the `relayout`, and `update` wrappers include full `supplyDefaults` - with the exception of `relayout` when the edits object `aobj` has had everything stripped out of it and executed directly within `_relayout` - and THAT can happen if all keys are handled by `editContainerArray`, which is OK because inside `editContainerArray` we call `supplyDefaults` for that particular component array. This won't work for the regex arrays, because the array name doesn't match a component, but then we don't find a `draw` method either so it reverts to a full replot.

So I think the only potential pitfall of this is if a change inside the array affects something in the layout outside the array. One case I thought *might* fit this bill is if you provide a new axis id in a component object (new or edited) - but actually out now as previously we don't generate new axes just from component references, so:

`Plotly.relayout(gd,{'annotations[1]':{text:'boo',x:-1,y:4, xref:'x2', yref:'y2'}})`

on a plot with only `xaxis` and `yaxis` will get drawn on the existing axes... but the bug is that in this case the axes don't get correctly autoranged until you redraw the plot. It's ONLY this case, autorange works if your axis refs are valid.

So, are there any cases I'm not thinking of where changes inside one of these arrays affect `_fullLayout` outside of the array?